src/native/managed/cdac/mscordaccore_universal/Legacy/SOSDacImpl.cs

```diff
@@ -117,45 +117,41 @@ int ISOSDacInterface.GetAppDomainData(ClrDataAddress addr, DacpAppDomainData* da
         try
         {
             if (addr == 0)
-            {
-                hr = HResults.E_INVALIDARG;
-            }
-            else
-            {
-                *data = default;
-                data->AppDomainPtr = addr;
-                TargetPointer systemDomainPointer = _target.ReadGlobalPointer(Constants.Globals.SystemDomain);
-                ClrDataAddress systemDomain = _target.ReadPointer(systemDomainPointer).ToClrDataAddress(_target);
-                Contracts.ILoader loader = _target.Contracts.Loader;
-                TargetPointer globalLoaderAllocator = loader.GetGlobalLoaderAllocator();
-                data->pHighFrequencyHeap = loader.GetHighFrequencyHeap(globalLoaderAllocator).ToClrDataAddress(_target);
-                data->pLowFrequencyHeap = loader.GetLowFrequencyHeap(globalLoaderAllocator).ToClrDataAddress(_target);
-                data->pStubHeap = loader.GetStubHeap(globalLoaderAllocator).ToClrDataAddress(_target);
-                data->appDomainStage = DacpAppDomainDataStage.STAGE_OPEN;
-                if (addr != systemDomain)
-                {
-                    TargetPointer pAppDomain = addr.ToTargetPointer(_target);
-                    data->dwId = _target.ReadGlobal<uint>(Constants.Globals.DefaultADID);
-
-                    IEnumerable<Contracts.ModuleHandle> modules = loader.GetModuleHandles(
-                        pAppDomain,
-                        AssemblyIterationFlags.IncludeLoading |
-                        AssemblyIterationFlags.IncludeLoaded |
-                        AssemblyIterationFlags.IncludeExecution);
+                throw new ReturnHResultException(HResults.E_INVALIDARG);
```

Comment:
Can we change this to `throw new ArgumentException()`? That sets the right HR value for `E_INVALIDARG`.